src/main/java/mate/academy/springboot/datajpa/controller/ProductController.java

```diff
@@ -0,0 +1,76 @@
+package mate.academy.springboot.datajpa.controller;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.springboot.datajpa.dto.mapper.RequestDtoMapper;
+import mate.academy.springboot.datajpa.dto.mapper.ResponseDtoMapper;
+import mate.academy.springboot.datajpa.dto.request.ProductRequestDto;
+import mate.academy.springboot.datajpa.dto.response.ProductResponseDto;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.service.ProductService;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+    private final RequestDtoMapper<ProductRequestDto, Product> requestMapper;
+    private final ResponseDtoMapper<ProductResponseDto, Product> responseMapper;
+
+    public ProductController(ProductService productService,
+                             RequestDtoMapper<ProductRequestDto, Product> requestMapper,
+                             ResponseDtoMapper<ProductResponseDto, Product> responseMapper) {
+        this.productService = productService;
+        this.requestMapper = requestMapper;
+        this.responseMapper = responseMapper;
+    }
+
+    @PostMapping
+    public ProductResponseDto add(@RequestBody ProductRequestDto requestDto) {
+        Product product = productService.save(requestMapper.toModel(requestDto));
+        return responseMapper.toDto(product);
+    }
+
+    @GetMapping("/{id}")
+    public ProductResponseDto get(@PathVariable Long id) {
+        Product product = productService.get(id);
+        return responseMapper.toDto(product);
+    }
+
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
+        productService.delete(id);
+    }
+
+    @PutMapping("/{id}")
+    public ProductResponseDto update(@RequestBody ProductRequestDto requestDto,
+                                      @PathVariable Long id) {
+        Product product = requestMapper.toModel(requestDto);
+        product.setId(id);
+        return responseMapper.toDto(productService.save(product));
+    }
+
+    @GetMapping("/by-price")
+    public List<ProductResponseDto> getAllByPrice(@RequestParam BigDecimal from,
+                                           @RequestParam BigDecimal to) {
+        return productService.findAllByPriceBetween(from, to).stream()

```

Comment:
```suggestion
    public List<ProductResponseDto> findAllByPriceBetween(@RequestParam BigDecimal from,
                                           @RequestParam BigDecimal to) {
        return productService.findAllByPriceBetween(from, to).stream()
```
keep to consistency in naming